taskcluster/mozillavpn_taskgraph/transforms/beetmover.py

```diff
@@ -0,0 +1,108 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+import os.path
+
+from taskgraph.transforms.base import TransformSequence
+from taskgraph.util.schema import resolve_keyed_by
+
+transforms = TransformSequence()
+
+
+@transforms.add
+def add_beetmover_worker_config(config, tasks):
+    for task in tasks:
+        worker_type = task["worker-type"]
+        task_name = task["name"]
+        task_label = f"{worker_type}-{task_name}"
+        resolve_keyed_by(
+            task,
+            "bucket",
+            item_name=task_label,
+            **{"level": config.params["level"]},
+        )
+        bucket = task["bucket"]
+        # When we add balrog releases we will need to add an action to beetmoverscript
+        action = "push-to-nightly"
+        app_name = "vpn"
+        attributes = task["attributes"]
+        run_on_tasks_for = task["run-on-tasks-for"]
+        build_id = config.params["moz_build_date"]
+        build_type = attributes["build-type"]
+        build_os = os.path.dirname(build_type)
+        dependencies = task["dependencies"]
+        app_version = config.params["version"]
+        candidates_path = os.path.join(
+            "pub", app_name, "candidates", app_version, build_id, build_os
+        )
+        destination_paths = [candidates_path]
+        archive_url = (
+            "https://ftp.stage.mozaws.net/"
+            if config.params["level"] != 3
+            else "https://ftp.mozilla.org/"
+        )
+        task_description = f"This {worker_type} task will upload a {build_os} release candidate for {app_name} v{app_version} to {archive_url}{candidates_path}/"
+        branch = config.params["head_ref"]
+
+        def get_artifact_path(path):
+            # iscript turns .tar.gz files into signed .pkg files
+            if build_os == "macos":
+                return path.replace(".tar.gz", ".pkg")
```

Comment:
I'm confused about this. If `iscript` converts it to a `.pkg` file, then shouldn't the artifact we're beetmoving already be in this format?

Reply:
This is confusing. The artifact we are beetmoving is indeed a `.pkg` file. The upstream artifact to sign (from the build) is a `tar.gz` file, but the artifact produced by the iscript signing task (the one want to beetmove) is a `.pkg` file. I tried to see if I could get the artifacts from the upstream signing task from some attribute in the signing task, but I was not able to. Maybe the build should produce a `.pkg` file? Not sure... iscript seems to expect a `tar.gz` file.

Reply:
[I was thinking I could use the `release-artifacts` but that's also suffixed `tar.gz`](https://github.com/mozilla-mobile/mozilla-vpn-client/blob/0fab5e0f5e00a6895313812148086dd4c0a1fb3f/taskcluster/ci/build/macos.yml#L24)